src/inapp/inapp.ts

```diff
@@ -420,7 +424,7 @@ export function getInAppMessages(
               values. This button is just a quality-of-life feature so that the customer will
               have an easy way to close the modal outside of the other methods.
             */
-            if (payload.closeButton) {
+            if (payload.closeButton && !isSafari) {
```

Comment:
Do not show close button if Safari because it won't work (Safari blocks JS on bound event handlers)